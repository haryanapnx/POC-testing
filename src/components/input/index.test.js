import * as React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Input from "./index";

describe("Input", () => {

  test("Snapshot", () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  test("renders correctly with default type text", () => {
    render(<Input />);
    const input = screen.getByRole("textbox");

    expect(input).toBeInTheDocument();
  });

  test("renders placeholder", () => {
    render(<Input placeholder="type text here" />);
    const input = screen.getByPlaceholderText("type text here");

    expect(input).toBeInTheDocument();
  });

  test("input disabled", () => {
    render(<Input disabled placeholder="type text here" />);

    const input = screen.getByRole("textbox");
    expect(input).toBeDisabled();
  });

  test("input readOnly", () => {
    render(<Input readOnly placeholder="type text here" />);

    const input = screen.getByRole("textbox");
    expect(input).toHaveAttribute("readOnly");

    const value = "this is value";

    userEvent.type(input, value);
    expect(input).not.toHaveValue(value);
  });

  test("input required", () => {
    render(<Input required={true} placeholder="type text here" />);

    const input = screen.getByRole("textbox");
    expect(input).toBeRequired();
  });

  test("input value", () => {
    const handleChange = jest.fn();
    render(<Input onChange={handleChange} placeholder="type text here" />);

    const input = screen.getByRole("textbox");
    const value = "this is value";

    userEvent.type(input, value);
    expect(handleChange).toHaveBeenCalledTimes(value.length);
    expect(input).toHaveValue(value);
  });

});
