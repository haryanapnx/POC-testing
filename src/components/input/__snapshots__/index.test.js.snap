// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot 1`] = `
<div>
  <label />
  <input
    class="my-input"
  />
</div>
`;
